src/libs/sfdk/dockervirtualmachine.cpp

```diff
@@ -0,0 +1,494 @@
+    data.insert(Constants::BUILD_ENGINE_SHARED_TARGET, m_sharedPaths[VirtualMachinePrivate::SharedTargets].first);
+    data.insert(Constants::BUILD_ENGINE_SHARED_CONFIG, m_sharedPaths[VirtualMachinePrivate::SharedConfig].first);
+    data.insert(Constants::BUILD_ENGINE_SHARED_SRC, m_sharedPaths[VirtualMachinePrivate::SharedSrc].first);
+    data.insert(Constants::BUILD_ENGINE_SHARED_SSH, m_sharedPaths[VirtualMachinePrivate::SharedSsh].first);
+
+    data.insert(Constants::BUILD_ENGINE_SSH_PORT, m_reservedPortMappings[VirtualMachinePrivate::SshPort].first);
+    data.insert(Constants::BUILD_ENGINE_WWW_PORT, m_reservedPortMappings[VirtualMachinePrivate::WwwPort].first);
+
+    return data;
+}
+
+VirtualMachineInfo DockerVirtualMachinePrivate::virtualMachineInfoFromOutput(const QByteArray &output) const
+{
+    QJsonDocument document = QJsonDocument::fromJson(output);
+    VirtualMachineInfo info = {};
+
+    if (document.isNull())
+        return info;
+
+    QJsonObject labels = document.object();
+
+    auto parsePort = [=](VirtualMachinePrivate::ReservedPort which, quint16& port, quint16 guestPort) {
+        QString key = label(which);
+        if (labels.contains(key)) {
+            port = labels.value(key).toInt();
+            m_reservedPortMappings[which] = QPair<quint16, quint16>(port, guestPort);
+        }
+    };
+
+    parsePort(VirtualMachinePrivate::SshPort, info.sshPort, GUESTSSHPORT);
+    parsePort(VirtualMachinePrivate::WwwPort, info.wwwPort, GUESTWWWPORT);
+
+    auto parsePath = [=](VirtualMachinePrivate::SharedPath which, QString& path, const QString& guestPath) {
+        QString key = label(which);
+        if (labels.contains(key)) {
+            path = labels.value(key).toInt();
+            m_sharedPaths[which] = QPair<QString, QString>(path, guestPath);
+        }
+    };
+
+    parsePath(VirtualMachinePrivate::SharedSrc, info.sharedSrc, Constants::BUILD_ENGINE_GUEST_TARGET_PATH);
+    parsePath(VirtualMachinePrivate::SharedSsh, info.sharedSsh, Constants::BUILD_ENGINE_SHARED_SSH);
+    parsePath(VirtualMachinePrivate::SharedHome, info.sharedHome, Constants::BUILD_ENGINE_SHARED_HOME);
+    parsePath(VirtualMachinePrivate::SharedConfig, info.sharedConfig, Constants::BUILD_ENGINE_SHARED_CONFIG);
+    parsePath(VirtualMachinePrivate::SharedTargets, info.sharedTargets, Constants::BUILD_ENGINE_SHARED_TARGET);
+    return info;
+}
+
+
+void DockerVirtualMachinePrivate::fetchInfo(VirtualMachineInfo::ExtraInfos extraInfo,
+    const QObject *context, const Functor<const VirtualMachineInfo &, bool> &functor) const
+{
+    Q_Q(const DockerVirtualMachine);
+    Q_ASSERT(context);
+    Q_ASSERT(functor);
+
+    // TODO: fetch snapshot info if requested
+    Q_UNUSED(extraInfo)
+
+    QStringList arguments;
+    arguments.append(COMMAND_INSPECT);
+    arguments.append(q->name());
+    arguments.append(FORMAT_LABEL_LIST);
+
+    auto info = std::make_shared<VirtualMachineInfo>();
```

Comment:
In this class the info is built in single step, so it is ok to use an automatic variable scoped inside the lambda.

Reply:
fixed